invoices/invoiceregistry.go

```diff
@@ -1514,10 +1513,10 @@ func (i *InvoiceRegistry) SubscribeNotifications(
 	}
 	client.ntfnQueue.Start()
 
-	i.clientMtx.Lock()
+	// Always increment by 1 first, and our client ID will start with 1,
+	// not 0.
+	atomic.AddUint32(&i.nextClientID, 1)
```

Comment:
To really make this atomic, you need to take the return value of `atomic.AddUint32` as what's the new value. Otherwise the read access below might already take another value if some other goroutine is doing the exact same thing as @carlaKC mentioned.

Reply:
Fixed.